### PR TITLE
Fix Dockerfile

### PR DESCRIPTION
**Pull request by sylr**

### CODE REVIEW

File: Dockerfile.multi-stage [hunk 1]
```diff
@@ -3,9 +3,9 @@ FROM golang:1.12-alpine3.9 as builder
 ADD . $GOPATH/src/github.com/improbable-eng/thanos
 WORKDIR $GOPATH/src/github.com/improbable-eng/thanos
 
-RUN apk update && apk upgrade && apk add --no-cache alpine-sdk
+RUN apk update && apk upgrade && apk add --no-cache alpine-sdk bzr
```
Comment (bwplotka) [inline Dockerfile.multi-stage]: Does it needed still? I think we removed this requirement (:

Comment (sylr) [reply Dockerfile.multi-stage]: I believe it's sitll needed :(

```
sylvain@ubuntu-1604-dev:~/gopath/src/github.com/improbable-eng/thanos[dockerfile]$ make docker-multi-stage
>> building docker image 'thanos' with Dockerfile.multi-stage
Sending build context to Docker daemon  154.6MB
Step 1/9 : FROM golang:1.12-alpine3.9 as builder
1.12-alpine3.9: Pulling from library/golang
e7c96db7181b: Already exists
5297bd381816: Already exists
3a664477889c: Already exists
bb89ac7faaa9: Pull complete
8bb43bd9c21d: Pull complete
Digest: sha256:06ba1dae97f2bf560831497f8d459c68ab75cc67bf6fc95d9bd468ac259c9924
Status: Downloaded newer image for golang:1.12-alpine3.9
 ---> c7330979841b
Step 2/9 : ADD . $GOPATH/src/github.com/improbable-eng/thanos
 ---> a9a8a803650e
Step 3/9 : WORKDIR $GOPATH/src/github.com/improbable-eng/thanos
 ---> Running in 399764e0c031
Removing intermediate container 399764e0c031
 ---> 602274f37d7a
Step 4/9 : RUN apk update && apk upgrade && apk add --no-cache alpine-sdk
 ---> Running in 6caa1ec5802e
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
v3.9.4-1-g3c20033f75 [http://dl-cdn.alpinelinux.org/alpine/v3.9/main]
v3.9.3-94-g8dce0df69b [http://dl-cdn.alpinelinux.org/alpine/v3.9/community]
OK: 9766 distinct packages available
OK: 6 MiB in 15 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/38) Installing fakeroot (1.23-r0)
(2/38) Installing sudo (1.8.25_p1-r2)
(3/38) Installing libcap (2.26-r0)
(4/38) Installing pax-utils (1.2.3-r0)
(5/38) Installing openssl (1.1.1b-r1)
(6/38) Installing libattr (2.4.47-r7)
(7/38) Installing attr (2.4.47-r7)
(8/38) Installing tar (1.32-r0)
(9/38) Installing pkgconf (1.6.0-r0)
(10/38) Installing patch (2.7.6-r4)
(11/38) Installing libgcc (8.3.0-r0)
(12/38) Installing libstdc++ (8.3.0-r0)
(13/38) Installing lzip (1.20-r0)
(14/38) Installing nghttp2-libs (1.35.1-r0)
(15/38) Installing libssh2 (1.8.2-r0)
(16/38) Installing libcurl (7.64.0-r1)
(17/38) Installing curl (7.64.0-r1)
(18/38) Installing abuild (3.3.1-r0)
Executing abuild-3.3.1-r0.pre-install
(19/38) Installing binutils (2.31.1-r2)
(20/38) Installing libmagic (5.36-r0)
(21/38) Installing file (5.36-r0)
(22/38) Installing gmp (6.1.2-r1)
(23/38) Installing isl (0.18-r0)
(24/38) Installing libgomp (8.3.0-r0)
(25/38) Installing libatomic (8.3.0-r0)
(26/38) Installing mpfr3 (3.1.5-r1)
(27/38) Installing mpc1 (1.0.3-r1)
(28/38) Installing gcc (8.3.0-r0)
(29/38) Installing musl-dev (1.1.20-r4)
(30/38) Installing libc-dev (0.7.1-r0)
(31/38) Installing g++ (8.3.0-r0)
(32/38) Installing make (4.2.1-r2)
(33/38) Installing fortify-headers (1.0-r0)
(34/38) Installing build-base (0.5-r1)
(35/38) Installing expat (2.2.6-r0)
(36/38) Installing pcre2 (10.32-r1)
(37/38) Installing git (2.20.1-r0)
(38/38) Installing alpine-sdk (1.0-r0)
Executing busybox-1.29.3-r10.trigger
OK: 182 MiB in 53 packages
Removing intermediate container 6caa1ec5802e
 ---> d98cc8aa920a
Step 5/9 : RUN git update-index --refresh; make build
 ---> Running in 93eb1fdce9c8
Dockerfile.multi-stage: needs update
go: finding github.com/oklog/ulid v1.3.1
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/hashicorp/go-sockaddr v1.0.0
go: finding github.com/hashicorp/memberlist v0.1.3
go: finding github.com/grpc-ecosystem/go-grpc-prometheus v0.0.0-20181025070259-68e3a13e4117
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/fatih/structtag v1.0.0
go: finding github.com/armon/go-metrics v0.0.0-20180917152333-f0300d1749da
go: finding golang.org/x/oauth2 v0.0.0-20181203162652-d668ce993890
go: finding google.golang.org/appengine v1.4.0
go: finding github.com/NYTimes/gziphandler v1.1.1
go: finding github.com/Azure/azure-storage-blob-go v0.0.0-20181022225951-5152f14ace1c
go: finding github.com/opentracing/basictracer-go v1.0.0
go: finding github.com/go-kit/kit v0.8.0
go: finding github.com/gophercloud/gophercloud v0.0.0-20181206160319-9d88c34913a9
go: finding github.com/gogo/protobuf v1.2.0
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/opentracing/opentracing-go v1.0.2
go: finding github.com/olekukonko/tablewriter v0.0.1
go: finding github.com/prometheus/prometheus v0.0.0-20190328180107-4d60eb36dcbe
go: finding golang.org/x/net v0.0.0-20180724234803-3673e40ba225
go: finding github.com/opentracing-contrib/go-stdlib v0.0.0-20170113013457-1de4cc2120e7
go: finding github.com/sean-/seed v0.0.0-20170313163322-e2103e2c3529
go: finding github.com/mozillazg/go-cos v0.11.0
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/googleapis/gax-go v2.0.2+incompatible
go: finding github.com/fortytw2/leaktest v1.3.0
go: finding golang.org/x/sys v0.0.0-20181026203630-95b1ffbd15a5
go: finding github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
go: finding github.com/pascaldekloe/goe v0.0.0-20180627143212-57f6aae5913c
go: finding github.com/google/martian v2.1.0+incompatible
go: finding gopkg.in/fsnotify/fsnotify.v1 v1.4.7
go: finding github.com/lovoo/gcloud-opentracing v0.3.0
go: finding github.com/hashicorp/go-immutable-radix v1.0.0
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding gopkg.in/check.v1 v1.0.0-20161208181325-20d25e280405
go: finding github.com/miekg/dns v1.0.14
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-version v1.2.0
go: finding github.com/minio/minio-go v0.0.0-20190131015406-c8a261de75c1
go: finding github.com/dgrijalva/jwt-go v0.0.0-20161101193935-9ed569b5d1ac
go: finding github.com/StackExchange/wmi v0.0.0-20180725035823-b12b22c5341f
go: finding github.com/rlmcpherson/s3gof3r v0.5.0
go: finding golang.org/x/sys v0.0.0-20190322080309-f49334f85ddc
go: finding github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: finding github.com/mitchellh/go-testing-interface v1.0.0
go: finding github.com/influxdata/influxdb v0.0.0-20170331210902-15e594fc09f1
go: finding github.com/montanaflynn/stats v0.0.0-20180911141734-db72e6cae808
go: finding github.com/evanphx/json-patch v4.1.0+incompatible
go: finding github.com/shurcooL/httpfs v0.0.0-20171119174359-809beceb2371
go: finding contrib.go.opencensus.io/exporter/ocagent v0.4.1
go: finding github.com/sasha-s/go-deadlock v0.0.0-20161201235124-341000892f3d
go: finding github.com/VividCortex/ewma v1.1.1
go: finding github.com/go-ini/ini v1.21.1
go: finding github.com/cockroachdb/cockroach-go v0.0.0-20181001143604-e0a95dfd547c
go: finding gopkg.in/fsnotify/fsnotify.v1 v1.3.0
go: finding github.com/shopspring/decimal v0.0.0-20180709203117-cd690d0c9e24
go: finding github.com/google/pprof v0.0.0-20180605153948-8b03ce837f34
go: finding github.com/mwitkow/go-conntrack v0.0.0-20161129095857-cc309e4a2223
go: finding github.com/google/go-querystring v1.0.0
go: finding github.com/go-ole/go-ole v1.2.1
go: finding cloud.google.com/go v0.34.0
go: finding github.com/mitchellh/go-homedir v0.0.0-20180523094522-3864e76763d9
go: finding github.com/hashicorp/serf v0.0.0-20161007004122-1d4fa605f6ff
go: finding github.com/stretchr/testify v1.2.2
go: labix.org/v2/mgo@v0.0.0-20140701140051-000000000287: bzr branch --use-existing-dir https://launchpad.net/mgo/v2 . in /go/pkg/mod/cache/vcs/ca61c737a32b1e09a0919e15375f9c2b6aa09860cc097f1333b3c3d29e040ea8: exec: "bzr": executable file not found in $PATH
go: finding google.golang.org/grpc v1.17.0
go: finding golang.org/x/sys v0.0.0-20190124100055-b90733256f2e
go: finding github.com/stretchr/testify v1.3.0
go: finding github.com/prometheus/client_golang v0.9.2
go: finding github.com/julienschmidt/httprouter v0.0.0-20150905172533-109e267447e9
go: finding github.com/onsi/ginkgo v1.6.0
go: finding github.com/biogo/store v0.0.0-20160505134755-913427a1d5e8
go: finding github.com/certifi/gocertifi v0.0.0-20180905225744-ee1a9a0726d2
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding github.com/prometheus/common v0.0.0-20181218105931-67670fe90761
go: finding github.com/cenk/backoff v2.0.0+incompatible
go: finding google.golang.org/grpc v1.15.0
go: finding github.com/peterbourgon/diskv v0.0.0-20180312054125-0646ccaebea1
go: finding github.com/cespare/xxhash v1.1.0
go: finding github.com/grpc-ecosystem/grpc-gateway v1.6.3
go: finding github.com/hashicorp/go-msgpack v0.0.0-20150518234257-fa3f63826f7c
go: finding github.com/peterbourgon/g2s v0.0.0-20170223122336-d4e7ad98afea
go: finding github.com/petermattis/goid v0.0.0-20170504144140-0ded85884ba5
go: finding golang.org/x/text v0.3.1-0.20180807135948-17ff2d5776d2
go: finding k8s.io/client-go v2.0.0-alpha.0.0.20181121191925-a47917edff34+incompatible
go: finding github.com/hashicorp/memberlist v0.1.0
go: finding github.com/grpc-ecosystem/go-grpc-middleware v1.0.0
go: finding github.com/gopherjs/gopherjs v0.0.0-20181103185306-d547d1d9531e
go: finding github.com/jackc/fake v0.0.0-20150926172116-812a484cc733
go: finding github.com/hashicorp/go-sockaddr v0.0.0-20180320115054-6d291a969b86
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
go: finding github.com/facebookgo/clock v0.0.0-20150410010913-600d898af40a
go: finding github.com/oklog/oklog v0.0.0-20170918173356-f857583a70c3
go: finding github.com/miekg/dns v1.1.4
go: finding github.com/grpc-ecosystem/grpc-opentracing v0.0.0-20180507213350-8e809c8a8645
go: finding github.com/samuel/go-zookeeper v0.0.0-20161028232340-1d7be4effb13
go: finding gopkg.in/vmihailenco/msgpack.v2 v2.9.1
go: finding github.com/opentracing/opentracing-go v1.0.1
go: finding github.com/getsentry/raven-go v0.1.0
go: finding github.com/hashicorp/go-uuid v1.0.0
go: finding github.com/go-sql-driver/mysql v1.4.0
go: finding github.com/census-instrumentation/opencensus-proto v0.1.0
go: finding github.com/cockroachdb/cockroach v0.0.0-20170608034007-84bc9597164f
go: finding gopkg.in/fsnotify.v1 v1.4.7
go: finding github.com/prometheus/tsdb v0.6.1
go: finding github.com/onsi/gomega v1.4.1
go: finding golang.org/x/net v0.0.0-20190213061140-3a22650c66bd
go: finding github.com/prometheus/client_golang v0.9.1
go: finding github.com/smartystreets/goconvey v0.0.0-20180222194500-ef6db91d284a
go: finding github.com/julienschmidt/httprouter v1.1.0
go: finding github.com/golang/snappy v0.0.0-20160529050041-d9eb7a3d35ec
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding google.golang.org/api v0.0.0-20180910000450-7ca32eb868bf
go: finding gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: finding github.com/jmespath/go-jmespath v0.0.0-20160803190731-bd40a432e4c7
go: finding github.com/hashicorp/yamux v0.0.0-20181012175058-2f1d1f20f75d
go: finding github.com/hpcloud/tail v1.0.0
go: finding sigs.k8s.io/yaml v1.1.0
go: finding github.com/rubyist/circuitbreaker v2.2.1+incompatible
go: finding gopkg.in/inf.v0 v0.9.0
go: finding golang.org/x/crypto v0.0.0-20190128193316-c7b33c32a30b
go: finding github.com/mattn/go-runewidth v0.0.3
go: finding k8s.io/apimachinery v0.0.0-20181127025237-2b1284ed4c93
go: finding github.com/hashicorp/golang-lru v0.5.0
go: finding gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405
go: finding golang.org/x/net v0.0.0-20180906233101-161cd47e91fd
go: finding github.com/prometheus/client_model v0.0.0-20180712105110-5c3871d89910
go: finding github.com/googleapis/gnostic v0.0.0-20180520015035-48a0ecefe2e4
go: finding github.com/cockroachdb/apd v1.1.0
go: finding golang.org/x/text v0.3.0
go: finding github.com/aws/aws-sdk-go v0.0.0-20180507225419-00862f899353
go: finding github.com/shurcooL/vfsgen v0.0.0-20180711163814-62bca832be04
go: finding google.golang.org/api v0.1.0
go: finding github.com/mitchellh/reflectwalk v1.0.0
go: finding github.com/json-iterator/go v0.0.0-20180612202835-f2b4162afba3
go: finding gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: finding go.opencensus.io v0.18.1-0.20181204023538-aab39bd6a98b
go: finding github.com/pkg/errors v0.8.0
go: finding golang.org/x/crypto v0.0.0-20180621125126-a49355c7e3f8
go: finding github.com/hashicorp/consul v0.0.0-20180615161029-bed22a81e9fd
go: finding github.com/hashicorp/go-msgpack v0.5.3
go: finding github.com/jackc/pgx v3.2.0+incompatible
go: finding github.com/smartystreets/goconvey v0.0.0-20181108003508-044398e4856c
go: finding github.com/go-logfmt/logfmt v0.4.0
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c
go: finding github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf
go: finding github.com/google/gofuzz v0.0.0-20150304233714-bbcb9da2d746
go: finding github.com/knz/strtime v0.0.0-20181018220328-af2256ee352c
go: finding gopkg.in/yaml.v2 v2.2.1
go: finding github.com/Azure/azure-pipeline-go v0.1.8
go: finding github.com/hashicorp/go-cleanhttp v0.0.0-20160407174126-ad28ea4487f0
go: finding k8s.io/klog v0.1.0
go: finding go.opencensus.io v0.19.0
go: finding github.com/google/btree v0.0.0-20180124185431-e89373fe6b4a
go: finding github.com/olekukonko/tablewriter v0.0.0-20180912035003-be2c049b30cc
go: finding github.com/jtolds/gls v4.2.1+incompatible
go: finding github.com/smartystreets/assertions v0.0.0-20190116191733-b6c0e53d7304
go: finding github.com/codahale/hdrhistogram v0.0.0-20161010025455-3a0bb77429bd
go: finding github.com/Azure/azure-sdk-for-go v23.2.0+incompatible
go: finding github.com/mozillazg/go-httpheader v0.2.1
go: finding golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
go: finding github.com/prometheus/common v0.0.0-20181119215939-b36ad289a3ea
go: finding golang.org/x/net v0.0.0-20190125091013-d26f9f9a57f3
go: finding github.com/cockroachdb/cmux v0.0.0-20170110192607-30d10be49292
go: finding github.com/hashicorp/go-rootcerts v0.0.0-20160503143440-6bb64b370b90
go: launchpad.net/gocheck@v0.0.0-20140225173054-000000000087: bzr branch --use-existing-dir https://launchpad.net/~niemeyer/gocheck/trunk . in /go/pkg/mod/cache/vcs/f46ce2ae80d31f9b0a29099baa203e3b6d269dace4e5357a2cf74bd109e13339: exec: "bzr": executable file not found in $PATH
go: finding github.com/golang/groupcache v0.0.0-20180924190550-6f2cf27854a4
go: finding github.com/coreos/etcd v3.3.10+incompatible
go: finding github.com/elastic/gosigar v0.9.0
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/satori/go.uuid v1.2.0
go: finding golang.org/x/net v0.0.0-20181023162649-9b4f9f5ad519
go: finding github.com/golang/protobuf v1.2.0
go: finding golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
go: finding github.com/lightstep/lightstep-tracer-go v0.15.6
go: finding golang.org/x/crypto v0.0.0-20181029021203-45a5f77698d3
go: finding github.com/spf13/pflag v1.0.3
go: finding github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d
go: finding github.com/lib/pq v1.0.0
go: finding gopkg.in/ini.v1 v1.41.0
go: finding k8s.io/kube-openapi v0.0.0-20180629012420-d83b052f768a
go: finding google.golang.org/genproto v0.0.0-20180831171423-11092d34479b
go: finding golang.org/x/tools v0.0.0-20181023010539-40a48ad93fbe
go: finding github.com/Azure/go-autorest v11.2.8+incompatible
go: finding golang.org/x/time v0.0.0-20170424234030-8be79e1e0910
go: finding k8s.io/api v0.0.0-20181213150558-05914d821849
go: finding github.com/kr/pretty v0.1.0
go: finding github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7
go: finding github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742
go: finding github.com/oklog/run v1.0.0
go: finding github.com/fsnotify/fsnotify v1.4.7
go: error loading module requirements
make: *** [Makefile:207: go-mod-tidy] Error 1
The command '/bin/sh -c git update-index --refresh; make build' returned a non-zero code: 2
Makefile:125: recipe for target 'docker-multi-stage' failed
make: *** [docker-multi-stage] Error 2
```

Comment (GiedriusS) [reply Dockerfile.multi-stage]: What about this PR https://github.com/improbable-eng/thanos/pull/1137, then?

Comment (sylr) [reply Dockerfile.multi-stage]: Ok i found the problem. The go.sum file has not been cleaned correctly. I'll send a PR.

Comment (sylr) [reply Dockerfile.multi-stage]: I've added a commit which get rid of all bzr repo.

**Pull request merged**